File: Java/Solution02.java
public class Solution {
    public boolean Find(int target, int [][] array) {
        boolean flag = false;
        int m = array.length, n = array[0].length;
        int i = m - 1, j = 0;
        while (i >= 0 && j < n && !flag) {
            if (array[i][j] == target)
                flag = true;
            else if (array[i][j] < target)
                ++j;
            else
                --i;
        }
        return flag;
    }
}

File: C++/Solution02.cpp
// O(n)

class Solution {
public:
    void replaceSpace(char *str,int length) {
        int count = 0;
        for (int i = 0; i < length; ++i)
            if (str[i] == ' ')
                ++count;
        for (int i = length-1; i >= 0; --i) {
            if (str[i] != ' ')
                str[i+2*count] = str[i];
            else {
                str[i+2*count] = '0';
                str[i+2*count-1] = '2';
                str[i+2*count-2] = '%';
                --count;
            }
        }
    }
};
